File: algorithm/src/main/java/baekjoon/p02979/Main.java
package baekjoon.p02979;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt();
        int b = sc.nextInt();
        int c = sc.nextInt();
        int[][] inputs = new int[3][2];

        for (int i = 0; i < inputs.length; i++) {
            for (int j = 0; j < inputs[i].length; j++) {
                inputs[i][j] = sc.nextInt();
            }
        }

        System.out.println(solve(a, b, c, inputs));
    }

    private static int solve(int a, int b, int c, int[][] inputs) {
        int[] cnt = new int[101];
        for (int[] input : inputs) {
            fill(cnt, input);
        }

        int ret = 0;
        int[] set = {0, a, b, c};
        for (int n : cnt) {
            ret += (set[n] * n);
        }

        return ret;
    }

    private static void fill(int[] a, int[] b) {
        for (int i = b[0]; i < b[1]; i++) {
            a[i]++;
        }
    }

}

File: algorithm/src/main/java/baekjoon/p02979/Main.kt
package baekjoon.p02979

fun main(args: Array<String>) {
    val (a, b, c) = readLine()!!.split(" ").map { it.toInt() }
    val inputs = List(3) { readLine()!!.split(" ").map { it.toInt() } }
    println(solve(a, b, c, inputs))
}

private fun solve(a: Int, b: Int, c: Int, inputs: List<List<Int>>): Int {
    val cnt = IntArray(101)
    for (input in inputs) {
        for (i in input[0] until input[1]) {
            cnt[i]++
        }
    }

    var ret = 0
    val set = arrayOf(0, a, b, c)
    for (n in cnt) {
        ret += (n * set[n])
    }

    return ret
}